docker/main/Dockerfile

```diff
@@ -167,17 +166,25 @@ RUN apt-get -qq update \
 # Ensure python3 defaults to python3.9
 RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1
 
+# need to install pip for requirements
 RUN wget -q https://bootstrap.pypa.io/get-pip.py -O get-pip.py \
     && python3 get-pip.py "pip"
 
+RUN python3 -m pip install --upgrade pip setuptools wheel appdirs
+
+# install uv
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/
+
+# install required python deps
 COPY docker/main/requirements.txt /requirements.txt
-RUN pip3 install -r /requirements.txt
+RUN uv pip install --system --no-build-isolation -r /requirements.txt
 
 # Build pysqlite3 from source
 COPY docker/main/build_pysqlite3.sh /build_pysqlite3.sh
 RUN /build_pysqlite3.sh
 
 COPY docker/main/requirements-wheels.txt /requirements-wheels.txt
+# we still have to use pip to build wheel
```

Comment:
there are so many dependencies in requirements-wheels.txt, why is it necessary to use `pip` instead of `uv pip` here?

Reply:
agh I see, https://github.com/astral-sh/uv/issues/1681